src/common/urlUtils.ts

```diff
@@ -79,9 +132,11 @@ export async function fetch(url: string): Promise<string> {
     });
   }
 
-  const driver = url.startsWith('https://') ? https : http;
+  const isInsecure = url.startsWith('http://');
+  const driver = isInsecure ? http : https;
+  const rejectUnauthorized = !isInsecure && !(await isLoopback(url));
```

Comment:
nit: double negative on the `isInsecure`, would read easier if it was flipped to `isSecure`